### PR TITLE
refactor: adjusting the position of ai modules

### PR DESCRIPTION
**Pull request by tzssangglass**
### Description

Fixes # (issue)

### Checklist

- [x] I have explained the need for this PR and the problem it solves
- [x] I have explained the changes or the new features added to this PR
- [x] I have added tests corresponding to this change
- [ ] I have updated the documentation to reflect this change
- [x] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

File: apisix/http/route.lua [hunk 1]
```diff
@@ -21,7 +21,7 @@ local service_fetch = require("apisix.http.service").get
 local core = require("apisix.core")
 local expr = require("resty.expr.v1")
 local plugin_checker = require("apisix.plugin").plugin_checker
-local ai = require("apisix.core.ai")
+local internal_event = require("apisix.internal_event")
```
Comment (membphis) [inline apisix/http/route.lua:24]: I think `event` is a good name, we can remove `internal_`

File: apisix/http/route.lua [hunk 2]
```diff
@@ -92,7 +92,7 @@ function _M.create_radixtree_uri_router(routes, uri_routes, with_parameter)
         end
     end
 
-    ai.routes_analyze(uri_routes)
+    internal_event.push_event("routes_change", uri_routes)
```
Comment (membphis) [inline apisix/http/route.lua:95]: ditto

**Review by membphis — CHANGES_REQUESTED** [review]

File: apisix/http/route.lua (hunk 2 above)
Comment (membphis) [inline apisix/http/route.lua:95]: bad event name

how about `create_new_http_router`?

File: apisix/http/route.lua (hunk 1 above)
Comment (tzssangglass) [reply apisix/http/route.lua]: update

File: apisix/http/route.lua (hunk 2 above)
Comment (tzssangglass) [reply apisix/http/route.lua]: update

File: apisix/plugins/ai.lua [hunk 3]
```diff
@@ -24,10 +26,23 @@ local route_lrucache = core.lrucache.new({
     count = 512
 })
 
-local _M = {}
+local schema = {
+    type = "object",
+    properties = {
+    }
```
Comment (membphis) [inline apisix/plugins/ai.lua:32]: not allow any property

File: apisix/event.lua [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+function _M.push(type, ...)
+    local handler = events[type]
+    if handler then
+        handler(...)
+    end
+end
+
+function _M.register(type, handler)
+    events[type] = handler
```
Comment (membphis) [inline apisix/event.lua:30]: add todo comment

we can register more than one handler

**Review by membphis — APPROVED** [review]

File: apisix/event.lua (hunk 4 above)
Comment (tzssangglass) [reply apisix/event.lua:31]: done

File: apisix/http/route.lua [hunk 5]
```diff
@@ -92,7 +92,7 @@ function _M.create_radixtree_uri_router(routes, uri_routes, with_parameter)
         end
     end
 
-    ai.routes_analyze(uri_routes)
+    event.push("create_new_http_router", uri_routes)
```
Comment (soulbird) [inline apisix/http/route.lua:95]: It would be better to predefine the time name in the event.

**Pull request closed without merging**